beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -779,39 +786,56 @@ impl<E: EthSpec> AvailableBlock<E> {
         self.block.clone()
     }
 
-    pub fn blobs(&self) -> Option<&BlobSidecarList<E>> {
-        self.blobs.as_ref()
-    }
-
     pub fn blobs_available_timestamp(&self) -> Option<Duration> {
         self.blobs_available_timestamp
     }
 
-    pub fn data_columns(&self) -> Option<&DataColumnSidecarList<E>> {
-        self.data_columns.as_ref()
+    pub fn data(&self) -> &AvailableBlockData<E> {
+        &self.data
+    }
+
+    pub fn has_blobs(&self) -> bool {
+        match self.data {
+            AvailableBlockData::NoData => false,
+            AvailableBlockData::Blobs(_) => true,
```

Comment:
As mentioned earlier in the Rpc block path, we might end up using this variant for an empty list

https://github.com/sigp/lighthouse/blob/05727290fbef27df72e854de0ca9280ee6347101/beacon_node/network/src/sync/block_sidecar_coupling.rs#L117-L134

Reply:
Yeah this code is not good, we should fix it

Reply:
Added a check for `if num_expected_blobs == 0 { AvailableBlockData::NoData }` in both RPC paths